### PR TITLE
Feature/add proposal validation

### PR DESCRIPTION
**Pull request by ibarrasb**
## Description

- Implemented client-side validation for GrantRecipe and GrantProposal forms.  
- Added `ValidationService` (now instance-based) and Zod-based schemas to define required fields and rules.  
- Centralized constants and messages in `validation.ts` for easier updates.  
- Validation runs entirely on the client, blocking invalid submissions and allowing drafts to save.  
- Added comprehensive test coverage for both GrantRecipe and GrantProposal validation, including edge cases (trimming, rating boundaries, nested array paths, etc.).  
- Updated documentation to reflect instance-based usage (`validationService` instead of `ValidationService`) and to show import examples.  
-  **Documentation:** [Validation Guide](https://docs.google.com/document/d/1iuBbhaIoKDiprkNxfsrAUhEiJlJ9Dz9yfnCePV5KyXw/edit?usp=sharing)

## What type of PR is this? (check all applicable)

- [ ] Refactor  
- [x] Feature  
- [ ] Bug Fix  
- [ ] Optimization  

## Related Tickets & Documents
- Jira Story: [WAVE-42](https://digital-aid-seattle.atlassian.net/browse/WAVE-42)

### CODE REVIEW

File: src/services/validation/ValidationService.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  const out: Record<string, string> = {};
+  for (const issue of issues) {
+    const key = issue.path.join("."); // "description" or "inputParameters.0.key"
+    if (!out[key]) out[key] = issue.message;
+  }
+  return out;
+};
+
+export class ValidationService {
+  static validateGrantRecipe(data: GrantRecipeInput): ValidationResult {
```
Comment (jnakaso) [inline src/services/validation/ValidationService.ts]: let's remove static. It's easier to mock behavior in testing

**Review by jnakaso — CHANGES_REQUESTED** [review]
small requested change.

More importantly, let's add a some tests.  Not only would that verify the rules we added, but provide examples of usage.

**Comment by jnakaso** [comment]
BTW, when setting reviewers add the team digitalaidseattle/our-wave

**Review by jnakaso — APPROVED** [review]
Looks great.
